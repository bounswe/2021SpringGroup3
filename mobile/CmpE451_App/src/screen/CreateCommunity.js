import React, {useState, createRef} from 'react';
import {
  StyleSheet,
  TextInput,
  View,
  Text,
  KeyboardAvoidingView,
  Keyboard,
  ToastAndroid,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import {getToken} from '../services/asyncStorageService';
import {COLORS} from '../theme/colors';
import {TEXT, BASE_URL} from '../constants';
import {IconButton} from 'react-native-paper';

export default function CreateCommunity({navigation}) {
  const [name, setName] = useState('');
  const [iconUrl, setIconUrl] = useState('');

  const iconUrlInputRef = createRef();

  const handleCreateCommunity = async () => {
    Keyboard.dismiss();
    fetch(BASE_URL + 'communities', {
      method: 'POST',
      body: JSON.stringify({
        name: name,
        iconUrl: iconUrl,
      }),
      headers: {
        'Content-Type': 'application/json',
        'X-Platform': 'ANDROID',
        Authorization: await getToken(),
      },
    })
      .then(async response => {
        const status = response.status;
        response = await response.json();
        if (status === 201) {
          navigateMain();
        } else {
          ToastAndroid.show(response.message, ToastAndroid.SHORT);
        }
      })
      .catch(error => {
        console.info(error);
        ToastAndroid.show(TEXT.networkError, ToastAndroid.SHORT);
      });
  };

  const navigateMain = async () => {
    navigation.navigate('Home');
  };

  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <View style={styles.container}>
      <ScrollView
        keyboardShouldPersistTaps="handled"
        styles={styles.contentContainerStyle}>
        <IconButton
          icon="close"
          color="grey"
          size={20}
          onPress={() => navigation.navigate('Home')}
        />
        <View style={{alignItems: 'center'}}>
          <Text style={styles.header}>Create Community</Text>
        </View>
        <KeyboardAvoidingView enabled>
          <View style={styles.SectionStyle}>
            <TextInput
              style={styles.inputStyle}
              onChangeText={UserName => setName(UserName)}
              underlineColorAndroid="#f000"
              placeholder="Community Name"
              placeholderTextColor="#8b9cb5"
              autoCapitalize="sentences"
              returnKeyType="next"
              onSubmitEditing={() =>
                iconUrlInputRef.current && iconUrlInputRef.current.focus()
              }
              blurOnSubmit={false}
            />
          </View>
          <View style={styles.SectionStyle}>
            <TextInput
              style={styles.inputStyle}
              ref={iconUrlInputRef}
              onChangeText={IconUrl => setIconUrl(IconUrl)}
              underlineColorAndroid="#f000"
              placeholder="Icon URL"
              placeholderTextColor="#8b9cb5"
              returnKeyType="next"
              onSubmitEditing={() => Keyboard.dismiss()}
              blurOnSubmit={false}
            />
          </View>
          <TouchableOpacity
            style={styles.buttonStyle}
            activeOpacity={0.5}
            onPress={handleCreateCommunity}>
            <View>
              <Text style={styles.buttonTextStyle}>Create</Text>
            </View>
          </TouchableOpacity>
        </KeyboardAvoidingView>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 2,
  },
  header: {
    fontSize: 25,
    alignSelf: 'center',
    color: COLORS.textColor,
  },
  SectionStyle: {
    flexDirection: 'row',
    height: 40,
    marginTop: 20,
    marginLeft: 35,
    marginRight: 35,
    margin: 10,
  },
  buttonStyle: {
    backgroundColor: COLORS.buttonColor,
    borderWidth: 0,
    color: COLORS.buttonColor,
    borderColor: COLORS.buttonColor,
    height: 40,
    alignItems: 'center',
    borderRadius: 30,
    marginLeft: 35,
    marginRight: 35,
    marginTop: 20,
    marginBottom: 20,
  },
  buttonTextStyle: {
    color: COLORS.buttonTextColor,
    paddingVertical: 10,
    fontSize: 16,
  },
  inputStyle: {
    flex: 1,
    color: COLORS.inputTextColor,
    paddingLeft: 15,
    paddingRight: 15,
    borderWidth: 1,
    borderRadius: 30,
    borderColor: '#dadae8',
  },
  contentContainerStyle: {
    justifyContent: 'center',
    alignContent: 'center',
  },
  imageStyle: {
    width: '50%',
    height: 100,
    resizeMode: 'contain',
    margin: 30,
  },
});
