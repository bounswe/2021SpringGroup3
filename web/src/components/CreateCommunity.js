import React from 'react';
import { Row, Col, Form, Input, Typography, Button, Radio, Select, Card} from 'antd';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import { CreateCommunity as CreateCommunityRequest } from '../utils/helper';

const { Text, Title } = Typography;
const { Option } = Select;


const CreateCommunity = (props) => {
    const loginState = useSelector((state) => state.login);
    const navigate = useNavigate()
    const dispatch = useDispatch()

    const onFinish = async (values) => {
        console.log("Success: ", values);
        
        const id = await CreateCommunityRequest({name: values.communityName, token: loginState.token}, dispatch);
        navigate(`/communities/${id}`)
    }
    const onFinishFailed = (error) => {
        console.log("Failed: ",error);
    }

    const postCardStyle = {
        width: '100%',
        backgroundColor: '#f8f8f8',
        borderRadius: '20px'
      }

    return ( 
        <Card size="small" style={postCardStyle}>
            <Row>
                <Col span={24} align="middle">
                    <Title level={2} strong="true">Create a New Community</Title>
                </Col>
                <Col span={10} offset={7} align="middle">
                    <Form
                        name="basic"
                        onFinish={onFinish}
                        onFinishFailed={onFinishFailed}
                    >
                        <Text strong="true">Name</Text>
                        <Form.Item
                            name="communityName"
                            rules={[
                                {
                                    required:true,
                                    message: "Community name cannot be empty!"
                                }
                            ]}
                        > 
                            <Input placeholder="Community Name"/>
                        </Form.Item>
                        
                        <Text strong="true">Topics</Text>
                        <Form.Item
                            name="topics"
                            rules={[
                                { 
                                    required: true, message: 'Please choose at least one topic!', type: 'array' 
                                }
                            ]}
                        >
                            <Select mode="multiple">
                                <Option value="1">Topic 1</Option>
                                <Option value="2">Topic 2</Option>
                                <Option value="3">Topic 3</Option>
                            </Select>
                        </Form.Item>
                        
                        <Text strong="true">Description</Text>
                        <Form.Item
                            name="description"
                        >
                            <Input.TextArea/>
                        </Form.Item>

                        <Text strong="true">Community Type</Text>
                        <Form.Item 
                            name="communityType"
                            rules={[
                                {
                                    required:true,
                                    message: "Please choose the community type!"
                                }
                            ]}
                        >
                            <Radio.Group>
                            <Radio value="public">Public</Radio>
                            <Radio value="private">Private</Radio>
                            </Radio.Group>
                        </Form.Item>

                        <Form.Item>
                            <Button type="primary" shape="round" htmlType="submit">
                                Submit
                            </Button>
                        </Form.Item>
                    
                    </Form>
                </Col>
            </Row>         
        </Card>
     );
}
 
export default CreateCommunity;