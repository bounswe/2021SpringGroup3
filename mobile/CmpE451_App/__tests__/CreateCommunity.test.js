import React from 'react';
import CreateCommunity from '../src/screen/CreateCommunity';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<CreateCommunity />).toJSON();
  expect(tree).toMatchSnapshot();
});
