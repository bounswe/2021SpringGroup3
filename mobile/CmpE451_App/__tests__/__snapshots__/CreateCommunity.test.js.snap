// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00227b",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "padding": 7,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 22.5,
                "height": 45,
                "width": 45,
              },
              undefined,
              Object {
                "margin": 0,
              },
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "lineHeight": 30,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û±Åç
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Create Community
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 25,
              "marginRight": 20,
              "paddingBottom": 20,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3949ab",
                "marginVertical": 6,
              }
            }
          >
            Community Name
            :
          </Text>
          <TextInput
            autoCapitalize="sentences"
            blurOnSubmit={false}
            multiline={true}
            onChangeText={[Function]}
            placeholder="Community Name"
            placeholderTextColor="#8b9cb5"
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "#f5f5f5",
                "borderColor": "lightgray",
                "borderWidth": 1,
                "color": "black",
                "paddingBottom": 1,
              }
            }
            underlineColorAndroid="#f000"
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 25,
              "marginRight": 20,
              "paddingBottom": 20,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3949ab",
                "marginVertical": 6,
              }
            }
          >
            Community Photo
            :
          </Text>
          <TextInput
            blurOnSubmit={false}
            multiline={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Icon URL"
            placeholderTextColor="#8b9cb5"
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "#f5f5f5",
                "borderColor": "lightgray",
                "borderWidth": 1,
                "color": "black",
                "paddingBottom": 1,
              }
            }
            underlineColorAndroid="#f000"
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 25,
              "marginRight": 20,
              "paddingBottom": 20,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3949ab",
                "marginVertical": 6,
              }
            }
          >
            Community Description
            :
          </Text>
          <TextInput
            blurOnSubmit={false}
            multiline={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Description"
            placeholderTextColor="#8b9cb5"
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "#f5f5f5",
                "borderColor": "lightgray",
                "borderWidth": 1,
                "color": "black",
                "paddingBottom": 1,
              }
            }
            underlineColorAndroid="#f000"
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 25,
              "marginRight": 20,
              "paddingBottom": 20,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3949ab",
                "marginVertical": 6,
              }
            }
          >
            Private Community
            :
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "width": "80%",
                }
              }
            >
              Only members can view posts of the private communities
            </Text>
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="#81b0ff"
              style={
                Array [
                  Object {
                    "height": 31,
                    "width": 51,
                  },
                  Object {
                    "backgroundColor": "#3e3e3e",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#f4f3f4"
              tintColor="#767577"
              value={false}
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3849aa",
                "borderColor": "#3849aa",
                "borderRadius": 12,
                "borderWidth": 0,
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "opacity": 1,
                "width": "80%",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                    "paddingVertical": 10,
                    "textAlign": "center",
                  }
                }
              >
                CREATE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
